rewrite-maven/src/test/java/org/openrewrite/maven/AddDependencyTest.java

```diff
@@ -1699,6 +1700,236 @@ void addDependencyToAggregatingPomAndParentPomWhenAggregatingPomIsParent() {
         );
     }
 
+    @NullSource
+    @ParameterizedTest
+    @ValueSource(strings = { "compile"})
+    void addDependencyWithImplicitAndExplicitCompileScopeDoesNotChangeProvidedScope(String scope) {
+        rewriteRun(
+          spec -> spec.recipe(new AddDependency("jakarta.annotation", "jakarta.annotation-api", "2.1.1", null, scope, null, null, null, null, null, null, null)),
+          pomXml(
+            """
+              <project>
+                  <groupId>com.mycompany.app</groupId>
+                  <artifactId>my-app</artifactId>
+                  <version>1</version>
+                  <dependencies>
+                      <dependency>
+                          <groupId>jakarta.annotation</groupId>
+                          <artifactId>jakarta.annotation-api</artifactId>
+                          <version>2.1.1</version>
+                          <scope>provided</scope>
+                      </dependency>
+                  </dependencies>
+              </project>
+              """
+          )
+        );
+    }
+
+    @NullSource
+    @ParameterizedTest
+    @ValueSource(strings = { "compile"})
+    void addDependencyWithImplicitAndExplicitCompileScopeDoesNotChangeTransitiveProvidedScopeOverride(String scope) {
+        rewriteRun(
+          spec -> spec.recipe(new AddDependency("jakarta.annotation", "jakarta.annotation-api", "2.1.1", null, scope, null, null, null, null, null, null, null)),
+          mavenProject("root",
+            pomXml(
+              """
+                <project>
+                    <groupId>com.mycompany.app</groupId>
+                    <artifactId>root</artifactId>
+                    <version>1</version>
```

Comment:
Maybe our test harness is adding that automatically but you might be missing `<packaging>pom</packaging>` here.

Reply:
It's not necessary for parsing and proving the recipe works as intended, we try to keep the poms in these tests as minimal as possible, notice for example that there's also no namespace and schema attributes, nor the model version tag